src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = new Node<K, V>(key, value);
+        V valueByNewKey = getValue(key);
+        if (valueByNewKey != null) {
+            if (Objects.equals(value, valueByNewKey)) {
+                return;
+            }
+            putIntoTable(node, table);
+            return;
+        }
+        if (table == null || size == threshold) {
+            resize();
+        }
+        putIntoTable(node, table);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null || size == 0) {
+            return null;
+        }
+        int index = key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+        Node<K, V> current = table[index];
+        if (current == null) {
+            return null;
+        }
+
+        while (current != null) {
+            boolean areKeysEqual = Objects.equals(current.key, key);
+            if (areKeysEqual) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
```

Comment:
heavily over-engineered method. You can reuse existing method and be done in just a few lines like this:
``` 
        Node<K, V>[] oldTable = table;
        table = new Node[table.length * GROW_FACTOR];
        size = 0;
        threshold *= 2;
        for (Node<K, V> node : oldTable) {
            while (node != null) {
                put(node.key, node.value);
                node = node.next;
            }
        }
```